Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3489,55 +3492,40 @@ - (BOOL)shouldAnimateRowUpdates:(NSArray<YapDatabaseViewRowChange *> *)rowChange
     OWSAssert(rowChanges);
 
     // If user sends a new outgoing message, don't animate the change.
-    BOOL isOnlyInsertingNewOutgoingMessages = YES;
-    BOOL isOnlyUpdatingLastOutgoingMessage = YES;
-    NSNumber *_Nullable lastUpdateRow = nil;
-    NSNumber *_Nullable lastNonUpdateRow = nil;
+    BOOL isOnlyModifyingLastMessage = YES;
     for (YapDatabaseViewRowChange *rowChange in rowChanges) {
         switch (rowChange.type) {
             case YapDatabaseViewChangeDelete:
-                isOnlyInsertingNewOutgoingMessages = NO;
-                isOnlyUpdatingLastOutgoingMessage = NO;
-                if (!lastNonUpdateRow || lastNonUpdateRow.integerValue < rowChange.indexPath.row) {
-                    lastNonUpdateRow = @(rowChange.indexPath.row);
-                }
+                isOnlyModifyingLastMessage = NO;
                 break;
             case YapDatabaseViewChangeInsert: {
-                isOnlyUpdatingLastOutgoingMessage = NO;
                 ConversationViewItem *_Nullable viewItem = [self viewItemForIndex:(NSInteger)rowChange.finalIndex];
-                if ([viewItem.interaction isKindOfClass:[TSOutgoingMessage class]]
+                if (([viewItem.interaction isKindOfClass:[TSIncomingMessage class]] ||
+                        [viewItem.interaction isKindOfClass:[TSOutgoingMessage class]])
                     && rowChange.finalIndex >= oldViewItemCount) {
                     continue;
                 }
-                if (!lastNonUpdateRow || lastNonUpdateRow.unsignedIntegerValue < rowChange.finalIndex) {
-                    lastNonUpdateRow = @(rowChange.finalIndex);
-                }
+                isOnlyModifyingLastMessage = NO;
             }
             case YapDatabaseViewChangeMove:
-                isOnlyInsertingNewOutgoingMessages = NO;
-                isOnlyUpdatingLastOutgoingMessage = NO;
-                if (!lastNonUpdateRow || lastNonUpdateRow.integerValue < rowChange.indexPath.row) {
-                    lastNonUpdateRow = @(rowChange.indexPath.row);
-                }
-                if (!lastNonUpdateRow || lastNonUpdateRow.unsignedIntegerValue < rowChange.finalIndex) {
-                    lastNonUpdateRow = @(rowChange.finalIndex);
-                }
+                isOnlyModifyingLastMessage = NO;
                 break;
             case YapDatabaseViewChangeUpdate: {
-                isOnlyInsertingNewOutgoingMessages = NO;
-                ConversationViewItem *_Nullable viewItem = [self viewItemForIndex:(NSInteger)rowChange.finalIndex];
-                if (![viewItem.interaction isKindOfClass:[TSOutgoingMessage class]]
-                    || rowChange.indexPath.row != (NSInteger)(oldViewItemCount - 1)) {
-                    isOnlyUpdatingLastOutgoingMessage = NO;
+                if (rowChange.changes == YapDatabaseViewChangedDependency) {
```

Comment:
Important change: ignore "adjacent neighbor refreshes" when deciding whether or not to animate.